main/src/main/java/org/mobilitydata/gtfsvalidator/cli/Main.java

```diff
@@ -63,26 +64,22 @@ public static void main(String[] argv) {
         feedLoader.setNumThreads(args.getNumThreads());
         NoticeContainer noticeContainer = new NoticeContainer();
         GtfsFeedContainer feedContainer;
+        GtfsInput gtfsInput;
         try {
             if (args.getInput() == null) {
-                feedContainer = feedLoader.loadAndValidate(
-                        GtfsInput.createFromUrl(
-                                new URL(args.getUrl()),
-                                args.getStorageDirectory()),
-                        feedName,
-                        validatorLoader,
-                        noticeContainer);
+                if (Strings.isNullOrEmpty(args.getStorageDirectory())) {
+                    gtfsInput = GtfsInput.createFromUrlInMemory(new URL(args.getUrl()));
+                } else {
+                    gtfsInput = GtfsInput.createFromUrl(new URL(args.getUrl()), args.getStorageDirectory());
+                }
             } else {
-                feedContainer = feedLoader.loadAndValidate(
-                        GtfsInput.createFromPath(args.getInput()),
-                        feedName,
-                        validatorLoader,
-                        noticeContainer);
+                gtfsInput = GtfsInput.createFromPath(Paths.get(args.getInput()));
             }
         } catch (IOException | URISyntaxException | InterruptedException e) {
             e.printStackTrace();
             return;
         }
+        feedContainer = feedLoader.loadAndValidate(gtfsInput, feedName, validatorLoader, noticeContainer);
```

Comment:
Thanks, I did not have the time yet to remove the code duplication #574

Reply:
You are welcome! Now we have a clear case when code duplication should be avoided since there are three different ways to create GtfsInput.